### ISSUE TITLE
rootless: document using pam_cgfs

### ISSUE THREAD

**Issue by cyphar** [opencontainers/runc#1839]

**Comment by AkihiroSuda** [comment]
@cyphar 

Did you manage to get runc work with pam_cgfs?

With pam_cgfs, only subset of controllers (memory and freezer by default), but current runc doesn't seem to support `Apply`-ing subset of controllers.

https://github.com/opencontainers/runc/blob/a77c3195e91c3930b51092a52612cfe77576007b/libcontainer/cgroups/fs/apply_raw.go#L176-L186


To support pam_cgfs, probably we need to change `if isIgnorableError(m.Rootless, err) && m.Cgroups.Path == ""` to `if isIgnorableError(m.Rootless, err)`.

**Comment by cyphar** [comment]
No, we wouldn't want to drop `m.Cgroups.Path` because that would mean we would ignore explicit configurations that specify a cgroupPath.

I haven't had a chance to try to get `pam_cgfs` working with `runc` -- but it should (in principle) work. Each `sys.Apply` is done for a separate controller so permission errors in one controller won't affect the others. Not to mention that you can set `-a` as an option for `pam_cgfs` to allow delegation of all controllers.

I'll sit down and take a look at this later.

**Comment by AkihiroSuda** [comment]
Can we add a CLI flag like `runc run --cgroup-ignore-unwritable-controllers=cpu,devices....`?

**Comment by cyphar** [comment]
We could, but I'd really prefer we avoid that. Every flag we add to `runc` makes life harder for everyone.
